cpp/src/slice2swift/Gen.cpp

```diff
@@ -1381,12 +1383,70 @@ bool
 Gen::ServantExtVisitor::visitInterfaceDefStart(const InterfaceDefPtr& p)
 {
     const string swiftModule = getSwiftModule(p->getTopLevelModule());
+    const string servant = getRelativeTypeString(p, swiftModule);
+    const string unescapedName = removeEscaping(servant);
+    const string disp = unescapedName + "Disp";
+    const string traits = unescapedName + "Traits";
 
     out << sp;
     writeServantDocSummary(out, p, swiftModule);
-    out << nl << "extension " << getRelativeTypeString(p, swiftModule);
+    out << nl << "extension " << servant;
+    out << sb;
+    out << nl << "private static var defaultObject: " << getUnqualified("Ice.Object", swiftModule) << sb;
```

Comment:
We do create a new `Ice.ObjectI<NameTraits>` for each dispatch. This is indeed a little wasteful... but if you call over and over the `ice_` operations, this wastefulness is the least of your problems! 

Note that a protocol extension cannot define a stored property.